core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java

```diff
@@ -247,6 +247,9 @@ public static void parseField(FieldMapper.Builder builder, String name, Map<Stri
                 if (parserContext.isWithinMultiField()) {
                     throw new MapperParsingException("include_in_all in multi fields is not allowed. Found the include_in_all in field ["
                         + name + "] which is within a multi field.");
+                } else if (parserContext.indexVersionCreated().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED)) {
+                    throw new MapperParsingException("[include_in_all] is not allowed for indices created on or after version 6.0.0 as " +
+                                    "[_all] is deprecated.");
```

Comment:
I think it might be a useful to repeat the same message you added to `_all` field here. Something like `"As a replacement, you can use an [copy_to] on mapping fields to create your own catch all field."`